tracing-appender/src/builder.rs

```diff
@@ -0,0 +1,88 @@
+use std::sync::atomic::AtomicUsize;
+use time::OffsetDateTime;
+use crate::rolling::{create_writer_file, Inner, RollingFileAppender, Rotation};
+use crate::sync::RwLock;
+
+#[cfg(feature="compression")]
+use crate::compression::CompressionConfig;
+
+use crate::writer::WriterChannel;
+
+pub struct RollingFileAppenderBuilder {
+    log_directory: Option<String>,
+    log_filename_prefix: Option<String>,
+    rotation: Option<Rotation>,
+    next_date: Option<AtomicUsize>,
+    #[cfg(feature = "compression")]
+    compression: Option<CompressionConfig>
+}
+
+impl RollingFileAppenderBuilder {
+    pub fn new() -> Self {
+        RollingFileAppenderBuilder {
+            log_directory: None,
+            log_filename_prefix: None,
+            rotation: None,
+            next_date: None,
+            #[cfg(feature = "compression")]
+            compression: None
+        }
+    }
+
+    pub fn log_directory(mut self, log_directory: String) -> Self {
+        self.log_directory = Some(log_directory);
+        self
+    }
+
+    pub fn log_filename_prefix(mut self, log_filename_prefix: String) -> Self {
```

Comment:
Yes I will change it, the reason is that I was suggested by Inner having directory and filename_prefix as String field so I didn't want to accept a Path just to convert it to a String. For now I will change it to `AsRef<Path>`.

EDIT: I wasn't aware that Path is not sized which means it would have to be a reference in the builder struct.